src/librustc_codegen_llvm/builder.rs

```diff
@@ -202,9 +170,16 @@ impl BuilderMethods<'a, 'tcx> for Builder<'a, 'll, 'tcx> {
         v: &'ll Value,
         else_llbb: &'ll BasicBlock,
         num_cases: usize,
-    ) -> &'ll Value {
-        unsafe {
+        cases: impl Iterator<Item = (u128, &'ll BasicBlock)>,
```

Comment:
You can remove `num_cases` by requiring `ExactSizeIterator` instead of just `Iterator`.

Reply:
Done and also added `TrustedLen` as giving a wrong length to llvm is probably UB.